XExtensionItem/XExtensionItem.m

```diff
@@ -193,7 +193,22 @@ - (id)activityViewController:(UIActivityViewController *)activityViewController
         item.attachments = ({
             id activityItem = [self activityItemForActivityType:activityType];
             
-            NSItemProvider *mainAttachment = [[NSItemProvider alloc] initWithItem:activityItem typeIdentifier:self.typeIdentifier];
+            NSString *typeIdentifier = ^NSString *{
+                BOOL classHasChanged = ![[activityItem class] isSubclassOfClass:[self.placeholderItem class]];
```

Comment:
Do we care about the passed in identifier for any cases other than `NSData`? Could this be simplified by always taking the derived type except in the case where `NSData` is involved?